src/Access/Common/AuthenticationData.cpp

```diff
@@ -94,6 +100,44 @@ AuthenticationData::Digest AuthenticationData::Util::encodeSHA1(std::string_view
 }
 
 
+AuthenticationData AuthenticationData::makePasswordAuthenticationData(AuthenticationType type, String password)
```

Comment:
Quite strange function. Why does it support only sha256?

Reply:
It supports all the passwords type (and in the future will also support bcrypt), it is just SHA256 that needs special handling, which is creating salt.

Reply:
And in general, we need it to have a single place where the password authentication data is created, and not to copy and paste the code in 3 different places.